### PR TITLE
add liquidity and increase allowance

### PR DESCRIPTION
**Pull request by MessaoudAzidane**
### Description:
- Add Liquidity
- Add Liquidity cspr
- increase allowance (that's needed for tokens different than cspr).
- Some refactoring will be done after

### CODE REVIEW

File: src/routes/addLiquidity/controller.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Request, Response } from "express";
+import { AddLiquidityParams } from "./types";
+import { postAddLiquidityValidator } from "./validator";
+import { sendBadRequestResponse, sendOkResponse } from "../../utils";
+import { AddLiquidityService } from "./service";
+
+export const addLiquidity = async (req: Request, res: Response) => {
+  const addLiquidityParams = req.body as AddLiquidityParams;
```
Comment (specialistvlad) [inline src/routes/addLiquidity/controller.ts:8]: Should we have data validation here?

File: src/utils/types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Common Params for all endpoints
+
+export interface CommonOperationsPrams {
```
Comment (specialistvlad) [inline src/utils/types.ts:3]: typo

File: src/routes/addLiquidity/service.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+  const [tokenAPackageHash, tokenBPackageHash] = await Promise.all([
+    getTokenPackageHash(params.tokenA),
+    getTokenPackageHash(params.tokenB),
+  ]);
+  if (tokenAPackageHash == "" || tokenBPackageHash == "") {
+    throw "token not found";
+  }
+
+  switch (entryPoint) {
```
Comment (specialistvlad) [inline src/routes/addLiquidity/service.ts:79]: please separate to functions

File: src/routes/addLiquidity/service.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  if (tokenSymbol === "CSPR") tokenSymbol = "WCSPR";
+  const token = await tokensRepository.find({ where: { tokenSymbol } });
+
+  if (!token || token.length !== 1) {
+    return "";
+  }
+  return token[0].tokenAddress;
+};
+
+export const AddLiquidityService = async (params: AddLiquidityParams): Promise<any> => {
```
Comment (specialistvlad) [inline src/routes/addLiquidity/service.ts:63]: any?

**Review by mrrfid — COMMENTED** [review]
Will review once Vlad's questions are resolved.

File: src/routes/addLiquidity/controller.ts (hunk 1 above)
Comment (MessaoudAzidane) [reply src/routes/addLiquidity/controller.ts:8]: it right after

**Review by specialistvlad — APPROVED** [review]

**Review by mrrfid — APPROVED** [review]

**Pull request merged**